laika/downloader.py

```diff
@@ -253,38 +255,35 @@ def download_and_cache_file_return_first_success(url_bases, folder_and_file_name
     raise last_error
 
 
-def download_and_cache_file(url_base, folder_path, cache_dir, filename, compression='', overwrite=False):
-  folder_path_abs = os.path.join(cache_dir, folder_path)
+def download_and_cache_file(url_base, folder_path: str, cache_dir: str, filename: str, compression='', overwrite=False):
   filename_zipped = filename + compression
-
+  folder_path_abs = os.path.join(cache_dir, folder_path)
   filepath = str(hatanaka.get_decompressed_path(os.path.join(folder_path_abs, filename)))
-  filepath_zipped = os.path.join(folder_path_abs, filename_zipped)
+
   filepath_attempt = filepath + '.attempt_time'
 
   if os.path.exists(filepath_attempt):
     with open(filepath_attempt, 'rb') as rf:
       last_attempt_time = float(rf.read().decode())
     if time.time() - last_attempt_time < SECS_IN_HR:
       raise IOError(f"Too soon to try downloading {folder_path + filename_zipped} from {url_base} again since last attempt")
-
   if not os.path.isfile(filepath) or overwrite:
-    os.makedirs(folder_path_abs, exist_ok=True)
-
     try:
       data_zipped = download_file(url_base, folder_path, filename_zipped)
     except (IOError, pycurl.error, socket.timeout):
       unix_time = time.time()
-      tmp_dir = cache_dir + '/tmp'
-      os.makedirs(tmp_dir, exist_ok=True)
-      with tempfile.NamedTemporaryFile(delete=False, dir=tmp_dir) as fout:
-        fout.write(str.encode(str(unix_time)))
-      os.replace(fout.name, filepath + '.attempt_time')
+      with atomic_write(filepath_attempt, mode='wb') as wf:
+        wf.write(str.encode(str(unix_time)))
```

Comment:
```suggestion
      with atomic_write(filepath_attempt, mode='w') as wf:
        wf.write(str(unix_time))
```

Reply:
Thanks! I also cleaned up reading the attempt file